core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -991,7 +991,7 @@ private[spark] class TaskSetManager(
     for ((tid, info) <- taskInfos if info.running && info.executorId == execId) {
       val exitCausedByApp: Boolean = reason match {
         case exited: ExecutorExited => exited.exitCausedByApp
-        case ExecutorKilled | ExecutorDecommission(_) => false
+        case ExecutorKilled | ExecutorDecommission(_, _) => false
```

Comment:
I am wondering if we should instead pattern match in a separate arm like: 

```
_ @ ExecutorDecommission => false
```

To avoid having to change the case arms when we make changes to the structure definitions.